scripts/organize_action_curriculum.py

```diff
@@ -0,0 +1,208 @@
+                    shutil.copy(file, output_situation / f"infrared_{idx}.png")
+                # Depth Vis Files
+                for idx, file in enumerate(sorted(input_curriculum_dir.glob("depth_vis_*"))):
+                    shutil.copy(file, output_situation / f"depthvis_{idx}.png")
+                # Depth Files
+                for idx, depth_file in enumerate(file for file in sorted(input_curriculum_dir.glob("depth_*")) if "vis" not in file.stem):
+                    shutil.copy(depth_file, output_situation / f"depth_{idx}.png")
+                # Feature Files
+                # Fake Object Files
+                object_features = {"objects": [
+                    {
+                        "color": [0, 0, 0],
+                        "object_name": f"object{idx}",
+                        "sub_part": None,
+                        "subobject_id": '0',
+                        "texture": None,
+                        "distance": None,
+                        "viewpoint_id": cam,
+                        "stroke_graph": {
+                            "adjacency_matrix": [],
+                            "concept_name": object_name,
+                            "confidence_score": 1.0,
+                            "stroke_mean_x": 150.0,
+                            "stroke_mean_y": 100.0,
+                            "stroke_std": 2.0,
+                            "strokes_normalized_coordinates": []
+                        }
+                    }
+                    for idx, object_name in enumerate(itt.chain(["person"], objects_to_append.get(f"{args.input_split}_{action_debug_name}_{ex}_{cam}", [])))
+                ]}
+                with open(output_situation / "feature_0.yaml", "w", encoding="utf-8") as feature_obj_0:
+                    yaml.dump(object_features, feature_obj_0)
+                with open(output_situation / "feature_1.yaml", "w", encoding="utf-8") as feature_obj_1:
+                    yaml.dump(object_features, feature_obj_1)
+                # Action Feature copy
+                with open(action_feature_file, encoding="utf-8") as action_feature_file:
+                    action_features = yaml.safe_load(action_feature_file)
+                action_features["object_name"] = "0"
+                action_features["objects"] = [str(value+1) for value in range(len(objects_to_append.get(f"{args.input_split}_{action_debug_name}_{ex}_{cam}", [])))] if action_features["objects"] is None else action_features["objects"]
+                with open(output_situation / "action.yaml", "w", encoding="utf-8") as action_feature_out:
+                    yaml.dump(action_features, action_feature_out)
+
+                # Description file
+                with open(output_situation / "description.yaml", "w", encoding="utf-8") as description_file:
+                    yaml.dump({"language": f"a person {ACTION_DEBUG_NAME_TO_NAME.get(action_debug_name, action_debug_name)}"}, description_file)
```

Comment:
No change needed. This is probably for the best given we're using subset on actions where the other objects may not be visible. This does seem to mean we can't learn affordances with the current dataset?

Reply:
Ah! Good catch -- Actually the `m5_action` dataset was a version of this which encoded the first object in the annotated set (if it existed) as the object eg. `a person throws a ball` -- But I had just manually altered this generation. I should set this via a flag.